### ISSUE TITLE
OSError while running the sample code.

### ISSUE THREAD

**Issue by VASHISHT-RAHUL** [FalkonML/falkon#60]
Running the following command with GPU :

kernel = falkon.kernels.GaussianKernel(sigma=1, opt=options)
flk = falkon.Falkon(kernel=kernel, penalty=1e-5, M=5000, options=options)

gives OSError : /opt/conda/lib/python3.10/site-packages/falkon/c_ext/_C.so: undefined symbol: _ZNK5torch8autograd4Node4nameEv.
Please provide the solution at the earliest.

**Comment by Giodiro** [comment]
Hi Rahul,This is an installation error. How did you install falkon?Could you try to clone the repository and run `python setup.py develop` from inside it directly?Il 6 lug 2023 13:36, Rahul Vashisht ***@***.***> ha scritto:
Running the following command with GPU :
kernel = falkon.kernels.GaussianKernel(sigma=1, opt=options)
flk = falkon.Falkon(kernel=kernel, penalty=1e-5, M=5000, options=options)
gives OSError : /opt/conda/lib/python3.10/site-packages/falkon/c_ext/_C.so: undefined symbol: _ZNK5torch8autograd4Node4nameEv.
Please provide the solution at the earliest.

**Comment by elishaparhi** [comment]
Hi,
I also got the same error and it got resolved now. But now it is showing the below error:

OSError: libtorch_cuda_linalg.so: cannot open shared object file: No such file or directory

Please help me resolve this.

Thanks!

**Comment by Giodiro** [comment]
yeah that's something I haven't figured out just yet.If I didn't get the path of your installation wrong you should try to run`export LD_LIBRARY_PATH="/opt/conda/lib/python3.10/site-packages/torch/lib:$LD_LIBRARY_PATH"`And then run the sample code. No need to reinstall anything.Il 7 lug 2023 19:40, elishaparhi ***@***.***> ha scritto:
Hi,
I also got the same error and it got resolved now. But now it is showing the below error:
OSError: libtorch_cuda_linalg.so: cannot open shared object file: No such file or directory
Please help me resolve this.
Thanks!

**Comment by elishaparhi** [comment]
Thanks..this solved the problem.

I ran the below commands:

import os 
os.environ['LD_LIBRARY_PATH'] ='/opt/conda/lib/python3.10/site-packages/torch/lib:$LD_LIBRARY_PATH'
os.environ['LD_LIBRARY_PATH'] +=':/usr/lib64-nvidia'
os.environ['LD_LIBRARY_PATH'] +=':/usr/local/cuda-11.8/compat'

Now, there is another error:
File "/usr/local/lib/python3.10/dist-packages/falkon/utils/devices.py", line 92, in _get_gpu_device_info
    mem_free, mem_total = mem_get_info(g)
  File "/usr/local/lib/python3.10/dist-packages/falkon/c_ext/__init__.py", line 17, in call_cuda
    return getattr(torch.ops.falkon, name)(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/_ops.py", line 569, in __getattr__
    raise AttributeError(
AttributeError: '_OpNamespace' 'falkon' object has no attribute 'mem_get_info'

**Comment by elishaparhi** [comment]
Now, getting the below error:

File "/usr/local/lib/python3.10/dist-packages/falkon/utils/devices.py", line 92, in _get_gpu_device_info
mem_free, mem_total = mem_get_info(g)
File "/usr/local/lib/python3.10/dist-packages/falkon/c_ext/init.py", line 17, in call_cuda
return getattr(torch.ops.falkon, name)(*args, **kwargs)
File "/usr/local/lib/python3.10/dist-packages/torch/_ops.py", line 569, in getattr
raise AttributeError(
AttributeError: '_OpNamespace' 'falkon' object has no attribute 'mem_get_info'

Please help me resolve this.

**Comment by Giodiro** [comment]
Hi @elishaparhi 

Really sorry if this comes too late, but installation should be a lot easier now.. let me know if you try again and have further problems!

Giacomo
